core/src/avm2/activation.rs

```diff
@@ -2620,7 +2622,17 @@ impl<'a, 'gc> Activation<'a, 'gc> {
         let value2 = self.pop_stack();
         let value1 = self.pop_stack();
 
-        self.push_stack(value1 == value2);
+        if value1 == value2 {
+            self.push_stack(true);
+        } else {
+            if let Some(xml1) = value1.as_object().and_then(|obj| obj.as_xml_object()) {
+                if let Some(xml2) = value2.as_object().and_then(|obj| obj.as_xml_object()) {
+                    self.push_stack(E4XNode::ptr_eq(*xml1.node(), *xml2.node()));
+                    return Ok(FrameControl::Continue);
+                }
+            }
```

Comment:
Instead of adding this as a special case here, could we add this to Value's `PartialEq` implementation?

Reply:
This applies specifically to the strict_eq opcode - I don't think we'll necessarily want this in other places

Reply:
That's not necessarily true - in avmplus, Array `(last)indexof` uses `stricteq()` too. (and that's... the only exception apparently. But that's because `Vector.indexOf` is implemented in AS, so it implicitly already uses `stricteq` opcode).
That said, I'm also not convinced this should be in default `PartialEq` implementation.